source/Sagan-Core-Tests/InMemoryRepositoryProviderTest.class.st

```diff
@@ -12,5 +12,50 @@ InMemoryRepositoryProviderTest >> setUpRepositoryWith: aConflictCheckingStrategy
 
 	extraterrestrials := InMemoryRepositoryProvider new
 		createRepositoryStoringObjectsOfType: Extraterrestrial
+		checkingConflictsAccordingTo: aConflictCheckingStrategy.
+	ships := InMemoryRepositoryProvider new
+		createRepositoryStoringObjectsOfType: Spaceship
 		checkingConflictsAccordingTo: aConflictCheckingStrategy
 ]
+
+{ #category : #tests }
+InMemoryRepositoryProviderTest >> testSpaceshipQueryingForPositiveFirepower [
+
+	self setUpRepositoryWithNoConflictChecking.
+
+	ships store: ( Spaceship withoutFirepowerNamed: 'Solvalou' ).
+	ships store: ( Spaceship named: 'Solgrado' firepower: 7650 ).
+	ships store: ( Spaceship withUnlimitedFirepowerNamed: 'Zeodalley' ).
+
+	self
```

Comment:
It might be better to store the ships in a local collection then use `assertCollection: actual hasSameElements: expected` to avoid the size and then anySatisfy, also just checking for the name is not as good as checking that the object returned is the same.

Reply:
Asserting that the elements are the same is not exactly what I want, that's why I wrote the assertions like that. I was probably being too defensive and what you suggest might also work. I will try and see if it works.